### ISSUE TITLE
Process output 'env' contains only first array variable member

### ISSUE THREAD

**Issue by dennishendriksen** [nextflow-io/nextflow#1790]
### Expected behavior and actual behavior

The output variable is a List object with String values instead of a String containing only the first array variable member.

### Steps to reproduce the problem

```
#!/usr/bin/env nextflow
nextflow.enable.dsl=2

process getIndexedArray {
        output:
                env INDEXED_ARRAY
        script:
                """
                INDEXED_ARRAY=("Apple" "Orange")
                """
}

workflow {
        getIndexedArray | view
}
```

### Program output 

```
N E X T F L O W  ~  version 20.10.0
Launching `bash_indexed_array.nf` [intergalactic_raman] - revision: b54975aecf
executor >  local (1)
[c4/b26e71] process > getIndexedArray [100%] 1 of 1 ✔
Apple
```

### Environment 

* Nextflow version: 20.10.0 build 5430
* Java version: openjdk version "11.0.4" 2019-07-16
* Operating system: Linux
* Bash version: GNU bash, version 4.2.46(2)-release (x86_64-redhat-linux-gnu)

### Additional context

 .command.sh
```
#!/bin/bash -ue
INDEXED_ARRAY=("Apple" "Orange")

# capture process environment
set +u
echo INDEXED_ARRAY=$INDEXED_ARRAY > .command.env
```

.command.env
```
INDEXED_ARRAY=Apple
```

**Comment by dennishendriksen** [comment]
For associative arrays an error is reported:
```
#!/usr/bin/env nextflow
nextflow.enable.dsl=2

process getAssociativeArray {
        output:
                env ASSOCIATIVE_ARRAY
        script:
                """
                declare -A ASSOCIATIVE_ARRAY
                ASSOCIATIVE_ARRAY=([Apple]=1 [Orange]=2)
                """
}

workflow {
        getAssociativeArray | view
}
```

```
Error executing process > 'getAssociativeArray'

Caused by:
  Missing environment variable: ASSOCIATIVE_ARRAY
```

In this case I would have expected the output variable to be of type Map with String keys and String values.

**Comment by dennishendriksen** [comment]
Any chance of a comment on this issue? (@pditommaso possibly?)

**Comment by dennishendriksen** [comment]
This is still in issue

**Comment by pditommaso** [comment]
Solved by #2624
